src/gui/window/keyboard.rs

```diff
@@ -28,28 +37,169 @@ impl GuiKeyboard {
 
         for (i, key) in key_view.iter_visible_keys() {
             if !key.black {
-                let top_left = Pos2::new(map_x(key.left), top);
-                let bottom_right = Pos2::new(map_x(key.right), bottom);
-
-                let rect = Rect::from_min_max(top_left, bottom_right);
                 let color = colors[i].map(map_color).unwrap_or(Color32::WHITE);
 
-                mesh.add_colored_rect(rect, color)
+                if color == Color32::WHITE {
```

Comment:
I can see you re-used my `let color = colors[i].map(map_color).unwrap_or(Color32::WHITE);`

You don't have to change this, but ideally for something like this you would have something like:
```rs
if let Some(color) = colors[i].map(map_color) {
    .... note pressed code
} else {
    let color = Color32::WHITE;
    .... note unpressed code
}

Reply:
Right this is a better approach. Will change.